dask_ml/model_selection/_incremental.py

```diff
@@ -0,0 +1,288 @@
+    else:
+        y_test = yield client.scatter(y_test)
+    if isinstance(y_test, da.Array):
+        y_test = client.compute(y_test)
+    else:
+        y_test = yield client.scatter(y_test)
+
+    # Convert to batches of delayed objects of numpy arrays
+    X_train = X_train.to_delayed()
+    if hasattr(X_train, 'squeeze'):
+        X_train = X_train.squeeze()
+    y_train = y_train.to_delayed()
+    if hasattr(y_train, 'squeeze'):
+        y_train = y_train.squeeze()
+    X_train, y_train = dask.optimize(X_train.tolist(), y_train.tolist())
+
+    # Create order by which we process batches
+    # TODO: create a non-repetitive random and uniform ordering
+    order = list(range(len(X_train)))
+    rng.shuffle(order)
+    seen = {}
+    tokens = {}
+
+    def get_futures(time_step):
+        j = order[time_step % len(order)]
+
+        if time_step < len(order) and j not in seen:  # new future, need to tell scheduler about it
+            X_future = client.compute(X_train[j])
+            y_future = client.compute(y_train[j])
+            seen[j] = (X_future.key, y_future.key)
+
+            # This is a hack to keep the futures in the scheduler but not in memory
+            X_token = client.submit(len, X_future)
+            y_token = client.submit(len, y_future)
+            tokens[time_step] = (X_token, y_token)
+
+            return X_future, y_future
+
+        else:
+            x_key, y_key = seen[j]
+            return Future(x_key), Future(y_key)
+
+    # Submit initial partial_fit and score computations on first batch of data
+    X_future, y_future = get_futures(0)
+    for ident, model in models.items():
+        model = client.submit(_partial_fit, model, X_future, y_future, fit_params)
+        score = client.submit(_score, model, X_test, y_test, scorer)
+        models[ident] = model
+        scores[ident] = score
+
+    done = defaultdict(set)
+    seq = as_completed(scores.values(), with_results=True)
+    current_time_step = 0
+    next_time_step = current_time_step + 1
+    optimistic = set()  # set of fits that we might or might not want to keep
+    history = []
+
+    # async for future, result in seq:
+    while not seq.is_empty():
+        future, meta = yield seq.__anext__()
+        if future.cancelled():
+            continue
+        time_step = meta['time_step']
+        ident = meta['ident']
+
+        done[time_step].add(ident)
+        info[ident].update(meta)
+        history.append(meta)
+
+        # Evolve the model by a few time steps, then call score on the last one
+        model = models[ident]
+        for i in range(time_step, next_time_step):
+            X_future, y_future = get_futures(i + 1)
+            model = client.submit(_partial_fit, model, X_future, y_future,
+                                  fit_params, priority=-i + meta['score'])
+        score = client.submit(_score, model, X_test, y_test, scorer,
+                              priority=-time_step + meta['score'])
```

Comment:
Nice – the priority keyword is useful, and I wasn't aware of the implementation.

I should spend some time thinking about priority. I think we want to use a multiplicative ratio to try to remove scale: `priority=score / i`.